### PR TITLE
[PM-20] Add stats graphs

### PR DESCRIPTION
**Pull request by zahin-mohammad**
Implements Graph

### CODE REVIEW

**Comment by atulb98** [comment]
@zahin-mohammad can you do `[PM-20] <PR title>` for consistency with other PRs?

[PM-20]: https://mochahub.atlassian.net/browse/PM-20

File: app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import androidx.lifecycle.Observer
+import com.github.mikephil.charting.data.LineDataSet
+import kotlinx.android.synthetic.main.graph_fragment.*
+import com.github.mikephil.charting.data.LineData
+
+class GraphFragment : Fragment() {
+
+    companion object {
+        fun newInstance() = GraphFragment()
+    }
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt]: No parameters being passed, I don't think this is necessary here

File: app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.graph_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initGraph()
+        ContextCompat.getColor(this.context!!, R.color.GraphText);
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt]:
```suggestion
        ContextCompat.getColor(requireContext(), R.color.GraphText)
```

File: app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        graph.axisLeft.textSize = 15.0f
+        graph.axisLeft.textColor = ContextCompat.getColor(this.context!!, R.color.GraphText)
+
+        graph.axisRight.setDrawGridLines(false)
+        graph.axisRight.setDrawAxisLine(false)
+        graph.axisRight.isEnabled = false
+
+        graph.legend.isEnabled = false
+
+    }
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt]: You can use `apply` quite nicely here:

```kotlin
    private fun initGraph() {
        graph.apply {
            setDrawBorders(false)
            setDrawGridBackground(false)
            setDrawMarkers(false)
            description.text=""

            xAxis.setDrawAxisLine(false)
            xAxis.setDrawGridLines(false)
            xAxis.isEnabled = false

            axisLeft.setDrawGridLines(false)
            axisLeft.setDrawAxisLine(false)
            axisLeft.textSize = 15.0f
            axisLeft.textColor = ContextCompat.getColor(requireContext(), R.color.GraphText)

            axisRight.setDrawGridLines(false)
            axisRight.setDrawAxisLine(false)
            axisRight.isEnabled = false

            legend.isEnabled = false
        }
    }
```

File: app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        super.onViewCreated(view, savedInstanceState)
+        initGraph()
+        ContextCompat.getColor(this.context!!, R.color.GraphText);
+
+        viewModel.data.observe(viewLifecycleOwner, Observer {
+            val lineDataSet = LineDataSet(viewModel.data.value,"Power Score")
+            lineDataSet.mode = LineDataSet.Mode.HORIZONTAL_BEZIER
+            lineDataSet.lineWidth = 3.0f
+            lineDataSet.setDrawCircles(false)
+            lineDataSet.valueTextSize = 0.0f
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt]:
```kotlin
val lineDataSet = LineDataSet(it,"Power Score").apply {
                mode = LineDataSet.Mode.HORIZONTAL_BEZIER
                lineWidth = 3.0f
                setDrawCircles(false)
                valueTextSize = 0.0f
            }
```

File: app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            val lineData = LineData(lineDataSet)
+            graph.data = lineData
+            graph.invalidate(); // refresh
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt]: If you're gonna copy and paste, at least drop the semi-colons 😏 
```suggestion
            graph.invalidate() // refresh
```

File: app/src/main/java/io/mochahub/powermeter/graph/GraphViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package io.mochahub.powermeter.graph
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.github.mikephil.charting.data.Entry
+
+class GraphViewModel : ViewModel() {
+    val _data  = MutableLiveData<List<Entry>>(
```
Comment (brian-norman) [inline app/src/main/java/io/mochahub/powermeter/graph/GraphViewModel.kt]:
```suggestion
    private val _data  = MutableLiveData<List<Entry>>(
```

**Review by brian-norman — CHANGES_REQUESTED** [review]
Some Kotlin idiomatic changes but otherwise cool 😎 Mans gotta rebase

File: app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt (hunk 1 above)
Comment (zahin-mohammad) [reply app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt]: Resolved here https://github.com/power-meter/android-client/pull/13/commits/fb795ca5cdcf70ecc398ab013c63fef985d9e00e

File: app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt (hunk 3 above)
Comment (zahin-mohammad) [reply app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt]: Resolved here https://github.com/power-meter/android-client/pull/13/commits/5fcce41f41a3aade69fa244e131892b2bfd19695

File: app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt (hunk 4 above)
Comment (zahin-mohammad) [reply app/src/main/java/io/mochahub/powermeter/graph/GraphFragment.kt]: Resolved here https://github.com/power-meter/android-client/pull/13/commits/5fcce41f41a3aade69fa244e131892b2bfd19695

**Review by brian-norman — APPROVED** [review]
🚀 SKRRR

**Pull request merged**